### PR TITLE
integrate aie-pdi-transform into xclbinutil

### PR DESCRIPTION
**Pull request by xfreid**
#### Problem solved by the commit
Integrate aie-pdi-transform to xclbinutil per Larry's request

#### Bug / issue (if any) fixed, which PR introduced the bug, how it was discovered
N/A

#### How problem was solved, alternative solutions (if any) and why they were rejected
N/A

#### Risks (if any) associated the changes in the commit
N/A

#### What has been tested and how, request additional testing if necessary
Manual test

#### Documentation impact (if any)
None

### CODE REVIEW

**Comment by xfreid** [comment]
Per Larry's request, integrate aie-pdi-transform into xclbinutil
aie-pdi-transform git repo - https://gitenterprise.xilinx.com/ai-engine/aie-pdi-transform
I copied the relevant source files from this repo and put them in a sub-folder of xclbinutil
I modified the source files to fix all the warnings (in XRT build, all the warnings are treated as errors)
The source files in aie-pdi-transform don't need to be reviewed

File: src/runtime_src/tools/xclbinutil/CMakeLists.txt [hunk 1]
```diff
@@ -196,4 +196,10 @@ if (GTEST_FOUND)
 else()
   message (STATUS "GTest was not found, skipping generation of test executables")
 endif()
+
+# ==-- a i e - p d i - t r a n s f o r m --==================================================
+# aie-pdi-transform only supports linux build currently
+if(NOT WIN32)
+  add_subdirectory(aie-pdi-transform)
+endif()
```
Comment (dbenusov-xilinx) [inline src/runtime_src/tools/xclbinutil/CMakeLists.txt:204]: Looks good! That is what I was hoping for :)

**Review by dbenusov-xilinx — APPROVED** [review]

**Comment by dbenusov-xilinx** [comment]
@stsoe Adding you to take a look and merge the changes, since I lack the `p o w e r`. We need to add this to the xclbinutil tool repository.

**Review by stsoe — CHANGES_REQUESTED** [review]
Where does this code come from?  Why is it all legacy c-code?
What does it do, why, and how?
All sections in the PR description must be thoroughly answered before the PR can be reviewed.

**Comment by stsoe** [comment]
Closing while waiting for answers.

**Comment by xfreid** [comment]
The request was from Larry Liu and Sonal Santan
_"All current shipping xclbins need to be updated to use the new format. 
We need to integrate [aie-pdi-transform] (https://gitenterprise.xilinx.com/ai-engine/aie-pdi-transform) into xclbinutil so we can upgrade an existing xclbin into the new format using xclbinutil. The details of the tools are here—
https://confluence.xilinx.com/pages/viewpage.action?spaceKey=~huaj&title=Transform+PDI+In+IPU-PHX"_

aie-pdi-transform git repo - https://gitenterprise.xilinx.com/ai-engine/aie-pdi-transform
This code was developed and maintained by Hua, but I think he is no longer responsible for maintaining this repo. Since this is not my code, I don't know the background, we can ask Hua to explain why it is all legacy C code.

Per Sonal's guidance, we copied the source code from the repo above to XRT sot that transform_static can be built and shipped along with xclbinutil

The source code in aie-pdi-transform is going through legal sweep right now

@stsoe, I will forward you the relevant email thread

**Pull request closed without merging**